dialogue-core/src/main/java/com/palantir/dialogue/core/NodeSelectionStrategyChannel.java

```diff
@@ -93,6 +93,12 @@ static LimitedChannel create(Config cf, ImmutableList<LimitedChannel> channels)
 
     @Override
     public Optional<ListenableFuture<Response>> maybeExecute(Endpoint endpoint, Request request) {
+
+        Integer hostId = request.attachments().getOrDefault(RoutingAttachments.HOST_KEY, null);
+        if (hostId != null) {
+            return channels.get(hostId).maybeExecute(endpoint, request);
```

Comment:
we should verify that this works as expected across reloads. What if we use the delegate `LimitedChannel` itself as a key?

Reply:
How would per-host channel bind them then? all it knows are URLs and host indexes

Reply:
I think we can figure something else out for the per-host channel, it's a little too special -- I wouldn't worry about attaching it to the sticky implementation. Realistically I don't think it should have ever been made public.